### PR TITLE
Splits out Solidity tests

### PR DESCRIPTION
**Pull request by erak**
Counterpart of https://github.com/ethereum/solidity/pull/7475.

### CODE REVIEW

**Comment by leonardoalt** [comment]
Looks good. Should we wait for the respective Solidity PR to be merged first?

File: test/cli.js [hunk 1]
```diff
@@ -27,13 +17,13 @@ tape('CLI', function (t) {
   });
 
   t.test('no mode specified', function (st) {
-    var spt = spawn(st, './solcjs test/' + daodir + '/Token.sol');
+    var spt = spawn(st, './solcjs test/contracts/Smoke.sol');
```
[the remark this answers is not in the document]
Comment (erak) [reply test/cli.js]: Yes :)

**Comment by erak** [comment]
@leonardoalt I'd wait for the Solidity PR to be merged first and make sure that everything is working as expected. I tested the other PR without these changes, and it seems to work. I'll give an updated if there's a change in plans.

**Comment by leonardoalt** [comment]
@erak Ok, I agree

**Pull request merged**